### DISCUSSION TITLE
Hello All How can i load a file from Local in filement i have done from internet and assert file.

### DISCUSSION THREAD

**Discussion by Amitdygon** [sceneview/sceneform-android#440] [Q&A]
Here i Have add my code for server url  Please help me how can i load it from local storage.

  `URL("URL").openStream().use { inputStream: InputStream ->
            val inputStream = BufferedInputStream(inputStream)
            ByteArrayOutputStream().use { output ->
                inputStream.copyTo(output)
                val byteArr = output.toByteArray()
                val byteBuffer = ByteBuffer.wrap(byteArr)
                val rewound = byteBuffer.rewind()
                withContext(Dispatchers.Main) {
                    modelViewer.destroyModel()
                    modelViewer.loadModelGlb(rewound)
                    modelViewer.transformToUnitCube()
                }
            }
        }`

**Comment by Amitdygon** [comment]
Found the answer

**Reply by Pey** [reply]
Hi,
Write the answer for everyone to use

**Comment by Amitdygon** [comment]
` private fun readAssetFromLocal(): ByteBuffer {
        val input :InputStream = FileInputStream(File(// your file path ))
        val bytes = ByteArray(input.available())
        input.read(bytes)
        return ByteBuffer.wrap(bytes)
    }`

` private fun loadGLB() {
        val buffer = readAssetFromLocal()
        modelViewer.loadModelGlb(buffer)
        modelViewer.transformToUnitCube()
    }

`
